### ISSUE TITLE
[Help] Get File from thumbnail

### ISSUE THREAD

**Issue by momoDragon** [CaiJingLong/flutter_photo_manager#602]
Is there a way to get file from the thumbnail? 

tried this but doesn't work 

```
          Uint8List? list = await assetEntity.thumbDataWithSize(400, 400);
          File file = File.fromRawPath(list!);
```

**Comment by vishnukvmd** [comment]
You can create a new `File` object at a desired path, and call `[writeAsBytes](https://api.flutter.dev/flutter/dart-io/File/writeAsBytes.html)` with the thumbnail data.

**Comment by momoDragon** [comment]
Yea i tried that and it works, but seems like i would store double the files. one from the package itself when it creates thumbnails and then another copy of my own in the temp directory. Can it be a feature request to have a method like `getFileFromThumbnail() `?

**Comment by vishnukvmd** [comment]
You could perhaps delete the one you have generated after you've made use of it?

If you could explain your exact use case, it would make it easier to suggest a solution.

**Comment by momoDragon** [comment]
My use case is that i need `File` as input for google_ml_kit package for ML processing. 
My concern is not just the space but also the time taken for saving the file. it would definitely be faster if i can get access to the thumbnail file directly, rather than writing to file then reading it again. I am trying to optimize the process because there can be scenarios of 100s of images.

**Comment by AlexV525** [comment]
AFAIK the `InputImage` accept `Uint8List`.
![image](https://user-images.githubusercontent.com/15884415/134287419-4d245b78-8d91-44f9-a2cc-4eb7970c1ec3.png)

**Comment by momoDragon** [comment]
yes i am aware of that too but i am not sure about `InputImageData` https://pub.dev/documentation/google_ml_kit/latest/google_ml_kit/InputImageData-class.html. 
Particularly these 
`imageRotation → InputImageRotation
Image rotation degree.
final
inputImageFormat → InputImageFormat
Format of the input image.
final
planeData → List<InputImagePlaneMetadata>?
The plane attributes to create the image buffer on iOS. [...]
final`

**Comment by momoDragon** [comment]
Bytes do not have these data. And also `AssetEntity` might not have all of it either

**Comment by AlexV525** [comment]
You can use the `.file` getter at present for your request, though it didn't provide any thumbnail options.

**Comment by momoDragon** [comment]
> You can use the `.file` getter at present for your request, though it didn't provide any thumbnail options.

Yup i tried that too but the file is unnecessarily large, causing ML Kit results to be slow. So for my case, using the full file is not an option.

**Comment by AlexV525** [comment]
> tried this but doesn't work
> 
> ```
>           Uint8List? list = await assetEntity.thumbDataWithSize(400, 400);
>           File file = File.fromRawPath(list!);
> ```

Regarding the first question, how it behave when you said `doesn't work`?

**Comment by momoDragon** [comment]
```
   Uint8List? list = await assetEntity.thumbDataWithSize(400, 400);
      File file = File.fromRawPath(list!);

      print(file.path);
      print(await file.readAsBytes());
```

Output:
<img width="1223" alt="Screenshot 2021-09-22 at 1 42 17 PM" src="https://user-images.githubusercontent.com/40727575/134289252-716fadb0-5848-41d9-96ae-8ead7ad3542b.png">

**Comment by AlexV525** [comment]
Can you make a minimal repo that includes ML Kit and this plugin to reproduce this issue?

**Comment by TylerKerch** [comment]
Hello, were you able to ever get this fixed? I have the same use-case of wanting to use a file with ml kit but without the lengthy load times of .file with large files.

**Comment by deepesh-inoid** [comment]
use this to achieve 

Future<File?> saveThumbnailToFile(AssetEntity asset) async {
    try {
      final Uint8List? thumbData = await asset.thumbnailDataWithSize(
          const ThumbnailSize(50, 50),
          quality: 100,
          format: ThumbnailFormat.jpeg
      );

      if (thumbData == null) {
        print("asset id null${asset.id}");
        return null;
      }

      final tempDir = await getTemporaryDirectory();

      final String sanitizedId = asset.id.replaceAll(RegExp(r'[^\w\-]'), '_');
      final String fileName = 'thumb_$sanitizedId.jpg';

      final String filePath = path.join(tempDir.path, fileName);

      final file = File(filePath);
      await file.writeAsBytes(thumbData);

      return file;
    } catch (e) {
      print("asset id ${asset.id}");
      return null;
    }
  }

**Comment by WutangNailao** [comment]
I would like to propose adding the capability to directly save the retrieved thumbnailData to a specified file path. like `asset .saveThumbnailToFile(String path)`

Currently, the Uint8List data must be transferred through the Flutter channel. In batch and high-concurrency scenarios, the serialization and memory copy overhead introduced by the Flutter channel is non-negligible and can significantly impact performance.

Providing an option to write the thumbnail data directly to a file on the native side would help avoid unnecessary data transfer between native and Dart layers, reduce memory pressure, and improve overall performance.

If this approach is acceptable, I would be happy to submit a PR to implement this feature.
